File: src/main/java/com/aliiensmp/aliienResize/Economy/currencyOptions/AxQuestBoard.java
package com.aliiensmp.aliienResize.Economy.currencyOptions;

import com.aliiensmp.aliienResize.Economy.CurrencyProvider;
import com.artillexstudios.axquestboard.api.AxQuestBoardAPI;
import org.bukkit.Bukkit;
import org.bukkit.entity.Player;

public class AxQuestBoard implements CurrencyProvider {

    private final String suffix;

    /**
     * Creates the AxQuestBoard adapter.
     *
     * @param suffix suffix displayed beside prices
     */
    public AxQuestBoard(String suffix) {
        this.suffix = suffix;
    }

    /**
     * Rounds a configured price up to the minimum whole number of points required.
     *
     * @param amount configured price
     * @return required point amount
     */
    private int getRequiredPoints(double amount) {
        return Math.max(0, (int) Math.ceil(amount));
    }

    @Override
    public boolean isValid() {
        return Bukkit.getPluginManager().getPlugin("AxQuestBoard") != null;
    }

    @SuppressWarnings("ConstantConditions")
    @Override
    public boolean hasBalance(Player player, double amount) {
        if (!isValid()) return false;

        int playerBalance = AxQuestBoardAPI.getQuestPoints(player.getUniqueId()).join();
        return playerBalance >= getRequiredPoints(amount);
    }

    @SuppressWarnings("ConstantConditions")
    @Override
    public boolean withdraw(Player player, double amount) {
        if (!hasBalance(player, amount)) return false;

        return AxQuestBoardAPI.takeQuestPoints(player.getUniqueId(), getRequiredPoints(amount)).join();
    }

    @Override
    public String suffix() {
        return suffix;
    }
}

File: src/main/kotlin/com/aliiensmp/aliienResize/economy/currencyOptions/AxQuestBoard.kt
package com.aliiensmp.aliienResize.economy.currencyOptions

import com.aliiensmp.aliienResize.economy.CurrencyProvider
import com.artillexstudios.axquestboard.api.AxQuestBoardAPI
import org.bukkit.Bukkit
import org.bukkit.entity.Player
import kotlin.math.ceil

class AxQuestBoard(override val suffix: String) : CurrencyProvider {

    override val isValid: Boolean
        get() = Bukkit.getPluginManager().getPlugin("AxQuestBoard") != null

    /**
     * Rounds a configured price up to the minimum whole number of points required.
     */
    private fun getRequiredPoints(amount: Double): Int {
        return maxOf(0, ceil(amount).toInt())
    }

    override fun hasBalance(player: Player, amount: Double): Boolean {
        if (!isValid) return false

        val balance = AxQuestBoardAPI.getQuestPoints(player.uniqueId).join()
        return balance >= getRequiredPoints(amount)
    }

    override fun withdraw(player: Player, amount: Double): Boolean {
        if (!hasBalance(player, amount)) return false

        return AxQuestBoardAPI.takeQuestPoints(player.uniqueId, getRequiredPoints(amount)).join()
    }
}